lib/compiler-finder.ts

```diff
@@ -33,21 +33,48 @@ import urljoin from 'url-join';
 
 import {InstanceFetcher} from './aws';
 import {logger} from './logger';
+import {PropertyGetter, PropertyValue, Widen} from './properties.interfaces';
+import {CompilerProps, RawPropertiesGetter} from './properties';
+import AWS from 'aws-sdk';
+import {ClientOptionsHandler} from './options-handler';
+import {CompileHandler} from './handlers/compile';
 
 const sleep = promisify(setTimeout);
 
+export type CompilerFinderArguments = {
```

Comment:
Based on https://github.com/compiler-explorer/compiler-explorer/blob/main/app.js#L152-L164 and https://github.com/compiler-explorer/compiler-explorer/blob/main/app.js#L71-L99